refactor: 설정 화면 컴포즈 마이그레이션

ijh1298 opened the pull request:
## 📌 관련 이슈
- close #783 

## ✨ 변경 내용

설정 화면을 Fragment 기반에서 Compose UI로 전환하고 코드 구조를 개선했습니다.

1. Compose 전환
- Fragment 5개를 Compose Screen 3개로 전환

2. 아키텍처 개선
- 이벤트 처리 로직을 ViewModel에서 분리
- Toast 및 화면 이동 이벤트를 SharedFlow로 마이그레이션

3. 화면 전환 수정
- 기존 새로운 `SettingActivity`를 띄우던 방식 -> 컴포저블을 Navigation 하는 방식으로 수정

## 🎸 기타 참고 사항

https://github.com/user-attachments/assets/be2a6316-d6a4-4049-826e-306b3a7e4daf



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

# Release Notes

* **New Features**
  * Integrated settings screens into main app navigation flow with improved accessibility
  * Enhanced dialog management for logout, nickname editing, and account deletion

* **Improvements**
  * Modernized settings UI with streamlined event handling
  * Refined toolbar consistency across settings screens
  * Improved account management screen organization

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ijh1298 committed:
feat: 설정 화면 Compose UI 구현

ijh1298 committed:
feat: 계정 관리, 계정 탈퇴 Compose UI 구현

ijh1298 committed:
feat: 설정 화면 다이얼로그 추가

- 닉네임 수정 다이얼로그 구현
- 로그아웃 다이얼로그 구현
- 회원탈퇴 다이얼로그 구현

ijh1298 committed:
refactor: 설정 화면 프로필 사진 수정 기능 연동

ijh1298 committed:
feat: 설정 화면 닉네임 변경 기능 연동

ijh1298 committed:
refactor: 공용 툴바 적용

- `BadgeToolbar`를 삭제하고 공용 `DefaultToolbar`로 대체합니다.

ijh1298 committed:
feat: 설정 화면 Compose Navigation으로 전환

ijh1298 committed:
refactor: 설정 Toast, 화면 이동 이벤트 SharedFlow, Navigation 마이그레이션

ijh1298 committed:
feat: 설정 화면 버튼 기능 연동

- 내 응원팀 변경 화면 이동
- 공지사항, 문의하기 외부 링크 연결
- 오픈소스 라이선스 화면 이동

ijh1298 committed:
feat: 회원 탈퇴 Firebase Analytics 로깅 추가

ijh1298 committed:
feat: 설정 TopBar 타이틀 표시 추가

ijh1298 committed:
fix: Top-level route에서 SettingMain 제거

ijh1298 committed:
fix: 회원 탈퇴 재확인 질문 다이얼로그 추가

ijh1298 committed:
feat: 가입일 형식 yyyy.MM.dd로 변경

ijh1298 committed:
refactor: 패키지 이동 및 사용하지 않는 XML, Fragment 제거

ijh1298 committed:
refactor: SettingMainScreen 컴포저블 분리 및 리팩토링

ijh1298 committed:
refactor: SettingDeleteAccountScreen 컴포저블 분리 및 리팩토링

ijh1298 committed:
refactor: SettingAccountScreen 컴포저블 분리 및 리팩토링

ijh1298 committed:
refactor: Setting 컴포저블의 Dialog, EventHandler 분리

ijh1298 committed:
refactor: SettingEvent 모델 분리

ijh1298 committed:
refactor: 화면 전환 애니메이션 로직 분리 및 리팩토링

ijh1298 committed:
refactor: MainScreen의 TopBar 분기 로직 when 식으로 변경

ijh1298 commented:
@coderabbitai review

ijh1298 committed:
refactor: 사용하지 않는 import 제거

ijh1298 committed:
fix: SettingAccountScreen, SettingDeleteAccountScreen 스크롤 가능하게 수정

ijh1298 committed:
refactor: 사용하지 않는 import 제거 및 상수 위치 이동

- `com.google.firebase.crashlytics.internal.common.IdManager.DEFAULT_VERSION_NAME` import 제거
- 관련 `DEFAULT_VERSION_NAME` 상수를 파일 내 private 상수로 이동

ijh1298 committed:
refactor: 사용하지 않는 import 제거

ijh1298 committed:
fix: 사용하지 않는 SettingActivity 제거

ijh1298 committed:
Merge branch 'dev-an' into refactor/783

# Conflicts:
#	android/app/src/main/AndroidManifest.xml

ijh1298 committed:
Merge remote-tracking branch 'origin/dev-an' into refactor/783

# Conflicts:
#	android/app/src/main/java/com/yagubogu/ui/setting/component/model/MemberInfoItem.kt

ijh1298 committed:
refactor: 불필요한 import 제거

jiyuneel commented on android/app/src/main/java/com/yagubogu/ui/setting/component/SettingButton.kt:
```diff
@@ -0,0 +1,90 @@
+                    .padding(horizontal = 20.dp, vertical = 10.dp),
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.SpaceBetween,
+        ) {
```
Box로 감쌀 필요 없이 Row 하나만 있어도 될 것 같아요. Box로 했을 때, 위아래 있는 Spacer가 아무런 역할이 없지 않나요?!

ijh1298 replied:
터치 영역을 넓힐 의도였는데 `Box`의 사용 목적은 다른 거였네요!
`Spacer`를 지우고 `Column`으로 수정해봤습니다

[refactor: SettingButton 수정](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/79810319482aac896661ff3c91af1fc6cce58f1c)

jiyuneel commented on android/app/src/main/java/com/yagubogu/ui/setting/SettingAccountScreen.kt:
```diff
@@ -0,0 +1,85 @@
+    settingEvent: SettingEvent?,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier =
+            modifier
+                .fillMaxSize()
+                .background(Gray050)
+                .padding(20.dp)
+                .verticalScroll(state = rememberScrollState()),
+        verticalArrangement = Arrangement.spacedBy(20.dp),
+    ) {
```
`SettingButtonGroup`의 modifier에 정의하면 `Column`으로 감쌀 필요가 없어 보여요!

ijh1298 replied:
`SettingButtonGroup`으로 modifier를 옮기고 `Column`을 제거하면 흰색 내부 `SettingButtonGroup`이 `fillMaxSize` 되어 화면에 꽉차서 의도한 대로 되지 않는거 같아요..

jiyuneel replied:
아하 전부 흰색으로 꽉차는군요 좋습니당-!!

jiyuneel reviewed (changes requested):
설정 화면 마이그레이션 고생 많으셨습니당~~ 👏
코멘트 확인 부탁해여

ijh1298 committed:
feat: NavigationRoot 및 Route 추가

- `Route` sealed interface를 추가하여 최상위 경로(Main, Setting) 정의
- `NavigationRoot`를 도입하여 앱의 메인 진입점과 설정 화면 사이의 네비게이션 계층 분리
- `TopNavKey`를 `SettingNavKey`로 변경하고 관련 설정을 `SettingScreen`으로 이동
- `Navigator`에 `canGoBack()` 및 `clearStack()` 메서드 추가
- `MainActivity`에서 `MainScreen` 대신 `NavigationRoot`를 호출하도록 수정

ijh1298 committed:
refactor: MainScreen 및 SettingScreen의 Navigator 의존성을 람다 콜백으로 교체

ijh1298 committed:
fix: 설정 사용자 정보 조회 시점 변경

ijh1298 committed:
refactor: 설정 model 패키지 위치 변경 (component.model -> model)

ijh1298 committed:
refactor: SettingButton 수정

ijh1298 commented on android/app/src/main/java/com/yagubogu/ui/setting/component/SettingButton.kt:
```diff
@@ -0,0 +1,90 @@
+import com.yagubogu.ui.util.noRippleClickable
+
+@Preview(showBackground = true)
+@Composable
+fun SettingButton(
+    modifier: Modifier = Modifier,
+    text: String = "",
+    onClick: () -> Unit = {},
+) {
```
좋습니다~

[refactor: SettingButton 수정](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/79810319482aac896661ff3c91af1fc6cce58f1c)

ijh1298 commented on android/app/src/main/java/com/yagubogu/ui/setting/component/SettingButton.kt:
```diff
@@ -0,0 +1,90 @@
+                contentDescription = null,
+                modifier = Modifier.size(20.dp),
+                tint = Gray500,
+            )
+        }
+        Spacer(
+            modifier =
+                Modifier
+                    .height(10.dp)
+                    .fillMaxWidth(),
+        )
+    }
+}
+
+@Composable
+fun SettingButtonGroup(
+    modifier: Modifier = Modifier,
+    content: @Composable () -> Unit,
+) {
+    Column(
+        modifier =
+            modifier
+                .fillMaxWidth()
+                .background(White, RoundedCornerShape(12.dp)),
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        Spacer(modifier = Modifier.height(20.dp))
+        Column(verticalArrangement = Arrangement.spacedBy(20.dp)) {
+            content()
+        }
+        Spacer(modifier = Modifier.height(20.dp))
+    }
```
좋습니다

[refactor: SettingButton 수정](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/79810319482aac896661ff3c91af1fc6cce58f1c)

ijh1298 committed:
refactor: SettingDialog 제거 및 개별 화면에서 다이얼로그 상태 관리로 변경

ijh1298 commented on android/app/src/main/java/com/yagubogu/ui/setting/SettingAccountScreen.kt:
```diff
@@ -0,0 +1,85 @@
+        modifier = modifier,
+    )
+
+    SettingDialog(viewModel = viewModel)
```
변경했습니다!

Screen 컴포저블에서 어떤 Dialog를 쓰는지 바로 알기 쉬워졌네요👍

[refactor: SettingDialog 제거 및 개별 화면에서 다이얼로그 상태 관리로 변경](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/2f18cd20f004ae06c489bba4f223bdeee2daa562)

ijh1298 committed:
refactor: SettingAccountScreen 콜백 함수명 변경 (click suffix)

ijh1298 committed:
design: 설정, 계정 관리 Screen Padding 수정

ijh1298 committed:
refactor: SettingMainScreen uCrop 인텐트 생성 함수명 변경 및 토스트 함수 수정

ijh1298 committed:
fix: SettingEventHandler에서 MainActivity 캐스팅 시 안정성 추가

ijh1298 commented on android/app/src/main/java/com/yagubogu/ui/setting/SettingAccountScreen.kt:
```diff
@@ -0,0 +1,85 @@
+            modifier
+                .fillMaxSize()
+                .background(Gray050)
+                .padding(20.dp)
```
[design: 설정, 계정 관리 Screen Padding 수정](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/998fe2d3ce3198048ddee7eb9db1f8ed2128c336)

ijh1298 commented on android/app/src/main/java/com/yagubogu/ui/setting/SettingMainScreen.kt:
```diff
@@ -0,0 +1,339 @@
+            modifier
+                .fillMaxSize()
+                .background(Gray050)
+                .padding(20.dp)
```
[design: 설정, 계정 관리 Screen Padding 수정](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/998fe2d3ce3198048ddee7eb9db1f8ed2128c336)

ijh1298 commented on android/app/src/main/java/com/yagubogu/ui/setting/SettingAccountScreen.kt:
```diff
@@ -0,0 +1,85 @@
+@Composable
+private fun SettingAccountScreen(
+    onClickLogout: () -> Unit,
+    onClickDeleteAccount: () -> Unit,
```
[refactor: SettingAccountScreen 콜백 함수명 변경 (click suffix)](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/bd3cc03903c4b3cb55f055edae12b8e02d95f0e6)

ijh1298 commented on android/app/src/main/java/com/yagubogu/ui/setting/SettingMainScreen.kt:
```diff
@@ -0,0 +1,339 @@
+                UCrop.RESULT_ERROR -> {
+                    val cropError: Throwable? = result.data?.let { UCrop.getError(it) }
+                    Timber.e(cropError, "uCrop Error")
+                    context.showToast(context.getString(R.string.setting_edit_profile_image_crop_failed))
```
[refactor: SettingMainScreen uCrop 인텐트 생성 함수명 변경 및 토스트 함수 수정](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/df55425d4734b5547e7fad25dcfd2c22bef8c40e)

ijh1298 commented on android/app/src/main/java/com/yagubogu/ui/setting/SettingMainScreen.kt:
```diff
@@ -0,0 +1,339 @@
+        DEFAULT_VERSION_NAME
+    }
+
+private fun createUCropIntent(
```
[refactor: SettingMainScreen uCrop 인텐트 생성 함수명 변경 및 토스트 함수 수정](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/df55425d4734b5547e7fad25dcfd2c22bef8c40e)

ijh1298 commented on android/app/src/main/java/com/yagubogu/ui/setting/SettingMainScreen.kt:
```diff
@@ -0,0 +1,339 @@
+        ProfileImage(memberInfoItem.profileImageUrl, modifier = Modifier.size(80.dp))
+        Spacer(modifier = Modifier.height(30.dp))
+        Text(text = memberInfoItem.nickName, style = PretendardSemiBold, fontSize = 24.sp)
+        Spacer(modifier = Modifier.height(16.dp))
```
👍👍
[design: 설정, 계정 관리 Screen Padding 수정](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/998fe2d3ce3198048ddee7eb9db1f8ed2128c336)

ijh1298 commented on android/app/src/main/java/com/yagubogu/ui/setting/component/SettingEventHandler.kt:
```diff
@@ -0,0 +1,60 @@
+            flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK
+        },
+    )
+    (context as MainActivity).finish()
```
좋습니다~~

[fix: SettingEventHandler에서 MainActivity 캐스팅 시 안정성 추가](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/2859fb67d552ea3b883ae8627501266e477f5fe8)

ijh1298 committed:
refactor: SettingScreen에서 불필요한 Box 레이아웃 제거

ijh1298 commented on android/app/src/main/java/com/yagubogu/ui/main/MainScreen.kt:
현재 다음 PR #808 에서 언급한 문제가 있긴 합니다만 적용해봤습니다!

[feat: NavigationRoot 및 Route 추가](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/8310de57d7618339e2d00cdc4c61000994c21a26)

ijh1298 committed:
docs: NavigationRoot.kt에 KDoc 주석 추가

ijh1298 reviewed (commented):
꼼꼼한 리뷰 감사합니다 👍👍
Navigation 구조 변경이 일어났는데 이 부분 괜찮은지 확인 부탁드립니다~

jiyuneel commented on android/app/src/main/java/com/yagubogu/ui/setting/component/SettingButton.kt:
```diff
@@ -0,0 +1,93 @@
+                    .padding(horizontal = 20.dp, vertical = 10.dp),
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.SpaceBetween,
+        ) {
```
`Column`의 content로 `Row` 하나만 있는데 감쌀 필요가 없을 거 같아요.
터치 영역을 넓힐 의도라면 `Row`의 modifier 속성의 순서를 고려하면 충분히 넓힐 수 있습니다!

이렇게 padding 전에 clickable을 두면 똑같은 효과가 있어용

```kotlin
Modifier
    .fillMaxWidth()
    .noRippleClickable { onClick() }
    .padding(horizontal = 20.dp, vertical = 10.dp),
```

ijh1298 replied:
<img width="1011" height="288" alt="Frame 1" src="https://github.com/user-attachments/assets/3aa80d37-bab3-40c3-818c-58f81bfede5e" />

`Column`을 제거하면 터치 영역이 오른쪽 사진만큼 줄지 않나요?!

jiyuneel replied:
Modifier는 순서대로 적용되기 때문에 전체 넓이(padding이 포함된 영역 전체)가 터치돼요!
```kotlin
Modifier
    .fillMaxWidth()                                  // 1. 먼저 가로를 가득 채움
    .noRippleClickable { onClick() }                 // 2. 전체 크기에 클릭 리스너 (터치 영역 = 전체)
    .padding(horizontal = 20.dp, vertical = 10.dp),  // 3. 그 안쪽으로 여백 (터치 영역 안쪽에 여백 생김)
``` 

https://github.com/user-attachments/assets/05c9cd0b-087f-4800-842f-1dfb4633693f

`background`를 넣어보면 더 확인하기 쉬울 것 같네용

```kotlin
Modifier
    .fillMaxWidth()
    .background(Color.Red)
    .noRippleClickable { onClick() }
    .padding(horizontal = 20.dp, vertical = 10.dp)
    .background(Color.Yellow),
``` 

<img width="536" height="380" alt="image" src="https://github.com/user-attachments/assets/46d1f1e7-95a2-4a4b-b0f1-b9c85af21195" />

ijh1298 replied:
오 지식이 늘었습니다 👍

[feat: SettingButton UI 구조 단순화 (Column 제거)](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/761bac968767321584776d99ab508874f6316478)

jiyuneel commented on android/app/src/main/java/com/yagubogu/ui/setting/component/SettingButton.kt:
```diff
@@ -0,0 +1,93 @@
+    Column(
+        modifier =
+            modifier
+                .fillMaxWidth()
+                .background(White, RoundedCornerShape(12.dp)),
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        Column(
+            verticalArrangement = Arrangement.spacedBy(20.dp),
+            modifier = Modifier.padding(vertical = 20.dp),
+        ) {
+            content()
+        }
+    }
```
여기도 마찬가지로 불필요한 Column 중첩으로 보여요!
```suggestion
    Column(
        modifier =
            modifier
                .fillMaxWidth()
                .background(White, RoundedCornerShape(12.dp))
                .padding(vertical = 20.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(20.dp),
    ) {
        content()
    }
```

ijh1298 replied:
👍👍

[refactor: SettingButton 불필요한 Column 제거](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/46c936a103b24b19270fc3ddfad35153ca3629fb)

jiyuneel commented on android/app/src/main/java/com/yagubogu/ui/setting/model/SettingDialogEvent.kt:
이제 필요 없어졌네요!

ijh1298 replied:
[refactor: 사용하지 않는 SettingDialogEvent 삭제](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/735aa5eadfa8aa28a0360ae216d3953183ce121d)

jiyuneel commented on android/app/src/main/java/com/yagubogu/ui/navigation/Route.kt:
```diff
@@ -0,0 +1,12 @@
+import androidx.navigation3.runtime.NavKey
+import kotlinx.serialization.Serializable
+
+sealed interface Route : NavKey {
+    @Serializable
+    data object BottomRoute : Route
+
+    @Serializable
+    data object SettingRoute : Route
+}
```
취향차이긴 한데 `sealed interface`를 통해 계층 구조를 나타냈으니, 하위 클래스에 `Route`를 안붙여도 되지 않을까 싶어요
`Route.Bottom`, `Route.Setting`만으로도 의미가 충분히 명확하다고 느껴서요!

ijh1298 replied:
좋습니다~

[refactor: Route 클래스 내 목적지 이름 변경 (BottomRoute -> Bottom, SettingRoute …](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/81ccc432f964261454f108dd16b0dd21022b05ed)

jiyuneel commented on android/app/src/main/java/com/yagubogu/ui/navigation/Navigator.kt:
```diff
@@ -11,26 +11,41 @@ import androidx.navigation3.runtime.NavKey
         } else {
             currentStack.removeLastOrNull()
         }
     }
```
`canGoBack` 함수를 활용할 수 있지 않을까요?!
```suggestion
    fun canGoBack(): Boolean = currentRoute != state.topLevelRoute

    fun goBack() {
        // If we're at the base of the current route, go back to the start route stack.
        if (!canGoBack()) {
            state.topLevelRoute = state.startRoute
        } else {
            currentStack.removeLastOrNull()
        }
    }
```

ijh1298 replied:
동일한 기능을 하는거 같네요 수정했습니다~

[fix: Navigator의 canGoBack 로직 수정](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/045ef0853ae277021437dc95238049f7f823e78d)

jiyuneel commented on android/app/src/main/java/com/yagubogu/ui/navigation/TransitionSpec.kt:
사용하지 않고 있는 것 같아요!

ijh1298 replied:
[refactor: 사용하지 않는 TransitionSpec 파일 삭제](https://github.com/woowacourse-teams/2025-yagu-bogu/pull/791/commits/056639125ee4f09715143aa8f89a7ec5d5b8d77d)

jiyuneel reviewed (changes requested):
간단한 코멘트 몇개만 확인해주세요!
Navigation 관련 코드도 확인했는데, 다음 PR 리뷰하면서 더 꼼꼼하게 보겠습니당

ijh1298 committed:
refactor: SettingButton 불필요한 Column 제거

ijh1298 committed:
refactor: 사용하지 않는 SettingDialogEvent 삭제

ijh1298 committed:
refactor: Route 클래스 내 목적지 이름 변경 (BottomRoute -> Bottom, SettingRoute -> Setting)

ijh1298 committed:
refactor: 사용하지 않는 TransitionSpec 파일 삭제

ijh1298 committed:
fix: Navigator의 canGoBack 로직 수정

ijh1298 reviewed (commented):
리뷰 반영했습니다~👍

터치 영역은 제가 했을 땐 줄어드는거 같은데 한번 확인부탁드려용

jiyuneel reviewed (approved):
수고 많으셨습니다!!!!👏 터치 영역 관련 코멘트 하나만 남겼어용

ijh1298 committed:
feat: SettingButton UI 구조 단순화 (Column 제거)

ijh1298 committed:
style: 불필요한 공백 제거
